### DISCUSSION TITLE
Docker: where are files stored?

### DISCUSSION THREAD

**Discussion by Jocky92** [pglombardo/PasswordPusher#1042] [Q&A] [answered]
Docker: where are the uploaded files stored?
is it possible to change upload file location?
do i have to delete the files manual or will PasswordPusher delete the files automatical when the link expire?

**Comment by pglombardo** [comment, marked as answer]
Hi @Jocky92,

> Docker: where are the uploaded files stored?

The default location for local storage is `./storage`.  Read [this documentation](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-file-pushes) for all of the details.

> is it possible to change upload file location?

Yes we support Azure Storage, Google Cloud Storage and Amazon S3 (including all compatible like Digital Ocean).  Details in the document linked above.

> do i have to delete the files manual or will PasswordPusher delete the files automatical when the link expire?

Files are deleted automatically and entirely when a push expires.  There is no retrieving them again from that point.  The filenames are even scrubbed from the audit logs.

![Screenshot 2023-04-20 at 15 54 36](https://user-images.githubusercontent.com/395132/233388821-a2dbc7dc-b3d7-4c01-837e-fdf96e6081d5.png)

I hope this helps!
